### PR TITLE
MINIFICPP-700: Add MSI Support via CPACK

### PR DESCRIPTION
**Pull request by phrocker**
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -199,7 +199,11 @@ if(USE_SYSTEM_ZLIB STREQUAL "OFF")
   message("Using bundled zlib")
 
   if (WIN32)
+  if (CMAKE_BUILD_TYPE MATCHES RelWithDebInfo OR CMAKE_BUILD_TYPE MATCHES Release)
```
Comment (arpadboda) [inline CMakeLists.txt]: What's wrong with the previous approach, which used build type == debug? 
With a quick grep I found only these 3 valid values (RelWithDebInfo, Release, Debug)

File: nanofi/examples/CMakeLists.txt [hunk 2]
```diff
@@ -57,6 +57,8 @@ else ()
     set(LINK_END_FLAGS "")
 endif ()
 
+if (NOT WIN32)
```
Comment (arpadboda) [inline nanofi/examples/CMakeLists.txt]: That means we don't compile nanofi examples for Win at all.
If that's the goal, I would prefer having the Win-specific flags (Wholearchive) removed, too.

**Comment by apiri** [comment]
reviewing

File: msi/LICENSE.txt [hunk 3]
```diff
@@ -0,0 +1,1633 @@
+
```
[the remark this answers is not in the document]
Comment (phrocker) [reply msi/LICENSE.txt]: ah yeah. agreed. I'll get the cpack integrated into appveyor.

[the remark this answers is not in the document]
Comment (apiri) [reply msi/LICENSE.txt]: Created https://issues.apache.org/jira/browse/MINIFICPP-708 for the licensing discussion

**Comment by apiri** [comment]
@phrocker could you please rebase and resolve the conflicts?

**Comment by apiri** [comment]
@phrocker  didn't want to lose track of this.  is this in a place where it is ready for another review?

**Comment by phrocker** [comment]
@apiri let me take a look at the appveyor failure. Seems that cpack fails there. Will investigate that further.  thanks.

**Comment by apiri** [comment]
hey @phrocker, just wanted to check on the status of this.  are there remaining changes you wanted to do or is it good for another verification?

File: CMakeLists.txt (hunk 1 above)
Comment (phrocker) [reply CMakeLists.txt]: There are more than that. I specifically want to exclude MinSizeRel from these cases.

**Comment by phrocker** [comment]
@apiri Should be good. I can't get cpack with install to work on appveyor but I'm not the only one. Will continue to take a look for a subsequent PR.

**Comment by apiri** [comment]
@phrocker Sounds good.  Sorry to miss this before the big one got merged.  Would you mind rebasing this and then I can take care of getting the merge on.

**Comment by phrocker** [comment]
@apiri the test failure is due to https://issues.apache.org/jira/browse/MINIFICPP-743 and something I'm looking to solve separately. It sporadically happens, but *should* be an easy fix.

**Comment by apiri** [comment]
@phrocker will revisit this today/tonight when I get back in front of my windows instance

**Comment by apiri** [comment]
reviewing

**Pull request closed without merging**